eng/tools/lint-diff/src/runChecks.ts

```diff
@@ -0,0 +1,106 @@
+import { join } from "path";
+import { exec, ExecException } from "node:child_process";
+
+import { getOpenapiType } from "./markdown-utils.js";
+import { getPathToDependency, AutorestRunResult } from "./util.js";
+
+const MAX_EXEC_BUFFER = 64 * 1024 * 1024;
+
+export async function runChecks(
+  path: string,
+  runList: Map<string, string[]>,
+): Promise<AutorestRunResult[]> {
+  const dependenciesDir = await getPathToDependency("@microsoft.azure/openapi-validator");
+  const result: AutorestRunResult[] = [];
+
+  for (const [readme, tags] of runList.entries()) {
+    const changedFilePath = join(path, readme);
+    console.log(`Linting ${changedFilePath}`);
+
+    // TODO: Move this into getRunList
+    let openApiType = await getOpenapiType(changedFilePath);
+
+    // From momentOfTruth.ts:executeAutoRestWithLintDiff
+    // This is a quick workaround for https://github.com/Azure/azure-sdk-tools/issues/6549
+    // We override the openapi-subtype with the value of openapi-type,
+    // to prevent LintDiff from reading openapi-subtype from the AutoRest config file (README)
+    // and overriding openapi-type with it.
+    let openApiSubType = openApiType;
+
+    // If the tags array is empty run the loop once but with a null tag
+    const coalescedTags = tags?.length ? tags : [null];
+    for (const tag of coalescedTags) {
+      let tagArg = tag ? `--tag=${tag} ` : "";
+
+      let autorestCommand =
+        `npm exec --no -- autorest ` +
+        `--v3 ` +
+        `--spectral ` +
+        `--azure-validator ` +
+        `--semantic-validator=false ` +
+        `--model-validator=false ` +
+        `--message-format=json ` +
+        `--openapi-type=${openApiType} ` +
+        `--openapi-subtype=${openApiSubType} ` +
+        `--use=${dependenciesDir} ` +
+        `${tagArg} ` +
+        `${changedFilePath}`;
+
+      console.log(`autorest command: ${autorestCommand}`);
+      const executionResult = await executeCommand(autorestCommand);
+      const lintDiffResult = {
+        autorestCommand,
+        rootPath: path,
+        readme,
+        tag: tag ? tag : "",
+        openApiType,
+        ...executionResult,
+      };
+      logAutorestExecutionErrors(lintDiffResult);
+      result.push(lintDiffResult);
+      console.log("Lint diff result length:", lintDiffResult.stdout.length);
+    }
+  }
+
+  return result;
+}
+
+export async function executeCommand(
```

Comment:
Replace with `.git/src/exec.js`

Reply:
Added to https://github.com/Azure/azure-rest-api-specs/issues/33302